src/Schema/Directives/CountDirective.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Illuminate\Database\Eloquent\Model;
+use Nuwave\Lighthouse\Schema\Values\FieldValue;
+use Nuwave\Lighthouse\Exceptions\DirectiveException;
+use Nuwave\Lighthouse\Support\Contracts\FieldResolver;
+use Nuwave\Lighthouse\Support\Contracts\DefinedDirective;
+
+class CountDirective extends BaseDirective implements FieldResolver, DefinedDirective
+{
+    /**
+     * Name of the directive as used in the schema.
+     *
+     * @return string
+     */
+    public function name()
+    {
+        return 'count';
+    }
+
+    public static function definition(): string
+    {
+        return /* @lang GraphQL */ <<<'SDL'
+"""
+Resolve the field through a count.
+"""
+directive @count(
+  """
+  The relationship which you want to run the count on.
+  """
+  relation: String
+
+  """
+  The model to run the count on.
+  """
+  model: String
+) on FIELD_DEFINITION
+SDL;
+    }
+
+    /**
+     * Makes a simple count query for the relation.
+     *
+     * @param FieldValue $value
+     * @return FieldValue
+     */
+    public function resolveField(FieldValue $value)
+    {
+        return $value->setResolver(
+            function (Model $model) {
```

Comment:
That would fail if used on root field.

```suggestion
            function ($root): int {
```

Reply:
Yeah I actually forgot to commit some changes. My bad :) I have added tests now.